src/main/ruby/core/file.rb

```diff
@@ -1046,7 +1046,7 @@ def self.stat(path)
   #  File.symlink("testfile", "link2test")   #=> 0
   def self.symlink(from, to)
     n = POSIX.symlink Rubinius::Type.coerce_to_path(from), Rubinius::Type.coerce_to_path(to)
-    Errno.handle if n == -1
+    Errno.handle_nfi if n == -1
```

Comment:
Maybe we should have `Errno.handle_nfi n`, which does `if n < 0`.

Reply:
I just kept the existing style.

Reply:
Yeah I know - it's an open question.

Reply:
For some calls like getpriority, `errno` should only be looked at if `ret == -1`, and I think for some other calls other return values than `-1` can indicate errors (e.g. getpwd() which returns `nil` or a String), so it seems easier the current way.